src/util.ts

```diff
@@ -255,4 +259,68 @@ export interface SynthesizedApp {
    * The construct metadata file (if exists).
    */
   readonly constructMetadata?: string;
+}
+
+export function parseImports(spec: string): ImportSpec {
+  const splitImport = spec.split(PREFIX_DELIM);
+
+  // k8s@x.y.z
+  // crd.yaml
+  // url.com/crd.yaml
+  if (splitImport.length === 1) {
+    return {
+      source: spec,
+    };
+  }
+
+  // crd:=crd.yaml
+  // crd:=url.com/crd.yaml
+  if (splitImport.length === 2) {
+    return {
+      moduleNamePrefix: splitImport[0],
+      source: splitImport[1],
+    };
+  }
+
+  throw new Error('Unable to parse import specification. Syntax is [NAME:=]SPEC');
+}
+
+export function deriveFileName(url: string) {
+  const devUrl = matchCrdsDevUrl(url);
+  let filename = undefined;
+
+  if (devUrl) {
+    const lastIndexOfSlash = devUrl.lastIndexOf('/');
+    const lastIndexOfAt = devUrl.lastIndexOf('@');
+    filename = devUrl.slice(lastIndexOfSlash+1, lastIndexOfAt);
+  } else {
+    const lastIndexOfSlash = url.lastIndexOf('/');
+    const lastIndexOfYaml = url.lastIndexOf('.yaml');
+    filename = url.slice(lastIndexOfSlash+1, lastIndexOfYaml);
+  }
+
+  if (!filename) {
+    filename = createHash('sha256');
```

Comment:
The hash needs to be comprised out of the URL...i.e:

```ts
hash = createHash('sha256');
hash.update(url)
filename = hash.digest('hex')
```

Also note that right now, the function return a `Hash` instance in this path, not a `string`...
You would have caught this if you'd mention the return value of the function in the signature :)

Reply:
Wow, thanks Eli, TIL 🙌